web-admin/src/features/organizations/settings/UserDomainAllowListItem.svelte

```diff
@@ -0,0 +1,97 @@
+<script lang="ts">
+  import {
+    createAdminServiceCreateWhitelistedDomain,
+    createAdminServiceListWhitelistedDomains,
+    createAdminServiceRemoveWhitelistedDomain,
+    getAdminServiceListWhitelistedDomainsQueryKey,
+  } from "@rilldata/web-admin/client";
+  import SettingsItemContainer from "@rilldata/web-admin/features/organizations/settings/SettingsItemContainer.svelte";
+  import {
+    getUserDomain,
+    userDomainIsPublic,
+  } from "@rilldata/web-admin/features/projects/user-invite/selectors";
+  import Switch from "@rilldata/web-common/components/forms/Switch.svelte";
+  import Label from "@rilldata/web-common/components/forms/Label.svelte";
+  import LoadingCircleOutline from "@rilldata/web-common/components/icons/LoadingCircleOutline.svelte";
+  import { queryClient } from "@rilldata/web-common/lib/svelte-query/globalQueryClient";
+
+  export let organization: string;
+
+  $: userDomain = getUserDomain();
+  $: isPublicDomain = userDomainIsPublic();
+
+  $: allowedDomains = createAdminServiceListWhitelistedDomains(organization);
+  $: domainAllowed = !!$allowedDomains.data?.domains?.find(
+    (d) => d.domain === $userDomain.data,
+  );
+
+  const allowDomainMutation = createAdminServiceCreateWhitelistedDomain();
+  const disallowDomainMutation = createAdminServiceRemoveWhitelistedDomain();
+  async function updateAllowedDomain() {
+    if (domainAllowed) {
+      await $disallowDomainMutation.mutateAsync({
+        organization,
+        domain: $userDomain.data,
+      });
+    } else {
+      await $allowDomainMutation.mutateAsync({
+        organization,
+        data: {
+          domain: $userDomain.data,
+          role: "viewer",
+        },
+      });
+    }
+
+    void queryClient.refetchQueries(
+      getAdminServiceListWhitelistedDomainsQueryKey(organization),
+    );
+  }
+</script>
+
+<!-- hide if user's domain is not public and no domains are added -->
+{#if $isPublicDomain.data || $allowedDomains.data?.domains?.length}
```

Comment:
We should hide if the domain _is_ public, right?

Reply:
Good point. The design changed to always show this part. So lemme update